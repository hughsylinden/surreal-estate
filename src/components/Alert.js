import "../styles/Alert.css";
import React from "react";
import PropTypes from "prop-types";

const Alert = ({ success, message }) => {
  let className = "";
  if (success) {
    className = "success-message";
  } else {
    className = "failure-message";
  }
  return message && <div className={className}>{message}</div>;
};

Alert.propTypes = {
  success: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
};

export default Alert;
