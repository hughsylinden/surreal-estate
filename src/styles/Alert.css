.success-message{
  border: solid 2px green;
  background-color: #deffdf;
  margin:5px;
}

.failure-message{
  border: solid 2px red;
  background-color: #ffdede;
  margin:5px;
}