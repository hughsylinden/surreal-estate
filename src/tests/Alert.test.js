import { render, screen } from "@testing-library/react";
import React from "react";
import Alert from "../components/Alert";

describe("Alert", () => {
  const mockMessage = {
    message: "mockmessage",
    success: true,
  };

  test("renders alert component", () => {
    const { asFragment } = render(
      <Alert success={mockMessage.success} message={mockMessage.message} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("alert message with className attributed rendered", () => {
    render(
      <Alert success={mockMessage.success} message={mockMessage.message} />
    );
    const messageElement = screen.getByText("mockmessage");
    expect(messageElement).toBeInTheDocument();
    expect(messageElement).toHaveClass("success-message");
  });
});
