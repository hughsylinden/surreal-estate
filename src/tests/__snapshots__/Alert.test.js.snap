// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders alert component 1`] = `
<DocumentFragment>
  <div
    class="success-message"
  >
    mockmessage
  </div>
</DocumentFragment>
`;
